watertap/core/tests/test_zero_order_siso.py

```diff
@@ -0,0 +1,179 @@
+        m.fs.unit.removal_frac_mass_solute[0, "C"].fix(0.3)
+
+        return m
+
+    @pytest.mark.unit
+    def test_private_attributes(self, model):
+        assert model.fs.unit._tech_type is None
+        assert model.fs.unit._has_recovery_removal is True
+        assert model.fs.unit._fixed_perf_vars == []
+        assert model.fs.unit._initialize is initialize_siso
+        assert model.fs.unit._scaling is calculate_scaling_factors_siso
+        assert model.fs.unit._get_Q is _get_Q_siso
+        assert model.fs.unit._stream_table_dict == {
+            "Inlet": model.fs.unit.inlet,
+            "Treated": model.fs.unit.treated}
+        assert model.fs.unit._perf_var_dict == {
+            "Solute Removal": model.fs.unit.removal_frac_mass_solute}
+
+    @pytest.mark.unit
+    def test_build(self, model):
+        assert isinstance(model.fs.unit.properties_in, WaterStateBlock)
+        assert isinstance(model.fs.unit.properties_treated, WaterStateBlock)
+
+        assert isinstance(model.fs.unit.inlet, Port)
+        assert isinstance(model.fs.unit.treated, Port)
+
+        assert isinstance(model.fs.unit.recovery_frac_mass_H2O, Var)
+        assert isinstance(model.fs.unit.removal_frac_mass_solute, Var)
+        assert len(model.fs.unit.removal_frac_mass_solute) == 3
+
+        assert isinstance(model.fs.unit.water_recovery_equation, Constraint)
+        assert len(model.fs.unit.water_recovery_equation) == 1
+        assert isinstance(model.fs.unit.solute_treated_equation, Constraint)
+        assert len(model.fs.unit.solute_treated_equation) == 3
+
+    @pytest.mark.unit
+    def test_degrees_of_freedom(self, model):
+        from idaes.core.util.model_statistics import unfixed_variables_set
+        [print(i) for i in unfixed_variables_set(model)]
+        assert degrees_of_freedom(model) == 0
+
+    @pytest.mark.component
+    def test_unit_consistency(self, model):
+        assert_units_consistent(model)
+
+    @pytest.mark.component
+    def test_scaling(self, model):
+        iscale.calculate_scaling_factors(model)
+
+        assert iscale.get_constraint_transform_applied_scaling_factor(
+            model.fs.unit.water_recovery_equation[0]) == 1e5
+        assert iscale.get_constraint_transform_applied_scaling_factor(
+            model.fs.unit.solute_treated_equation[0, "A"]) == 1e5
+        assert iscale.get_constraint_transform_applied_scaling_factor(
+            model.fs.unit.solute_treated_equation[0, "B"]) == 1e5
+        assert iscale.get_constraint_transform_applied_scaling_factor(
+            model.fs.unit.solute_treated_equation[0, "C"]) == 1e5
+
+    @pytest.mark.component
+    def test_initialization(self, model):
+        initialization_tester(model)
+
+    @pytest.mark.component
+    def test_solve(self, model):
+        results = solver.solve(model)
+
+        # Check for optimal solution
+        assert check_optimal_termination(results)
+
+    @pytest.mark.component
+    def test_solution(self, model):
```

Comment:
I got a failure when running this test_solution in my laptop: 
![Review_err1](https://user-images.githubusercontent.com/98901358/155017352-66f386ec-6fc6-4812-9ebb-34f92cb3642f.PNG)

Reply:
Did `test_solve` also fail?

Reply:
No, it works.

Reply:
@luohezhiming This looks like it is something on your end - the answer definitely should be 1000 if it solved correctly. You probably need to look at what is happening during the solver call step (and maybe copy all the important code into a separate script and run it outside of `pytest` so you have full control over it.